### PR TITLE
Add Variable Domain Computation

### PR DESCRIPTION
**Pull request by ccoffrin**
@mlubin, are there any obvious cases that I missed in the `variable_domain` function?

### CODE REVIEW

**Comment by ccoffrin** [comment]
Working to address https://github.com/lanl-ansi/PowerModels.jl/issues/535

File: src/core/relaxation_scheme.jl [hunk 1]
```diff
@@ -1,12 +1,46 @@
+
+"""
+Computes the valid domain of a given JuMP varibale taking into account bounds
```
Comment (mlubin) [inline src/core/relaxation_scheme.jl]: s/varibale/variable/

File: src/core/relaxation_scheme.jl [hunk 2]
```diff
@@ -1,12 +1,46 @@
...
+and the varaible's implicit bounds (e.g. binary).
+"""
+function variable_domain(var::JuMP.VariableRef)
```
Comment (mlubin) [inline src/core/relaxation_scheme.jl]: This doesn't check for semi-infinite and semi-continuous sets. However, it's not trivial to detect if these constraints exist before JuMP 0.20. Maybe worth a note/todo.

Comment (ccoffrin) [reply src/core/relaxation_scheme.jl]: Will do.  Thanks for the review!

**Comment by ccoffrin** [comment]
FYI @tasseff, @kaarthiksundar take note of this new `variable_domain` function, which can be used to make bound-based relaxations more robust.  For a problematic example see, https://github.com/lanl-ansi/PowerModels.jl/issues/535.

**Pull request merged**